### PR TITLE
Added tests for shape function

### PR DESCRIPTION
**Pull request by parth121101**
Fixes: #3576

### CODE REVIEW

File: integration_tests/intrinsics_172.f90 [hunk 1]
```diff
@@ -0,0 +1,6 @@
+program intrinsics_172
+    print *, shape([[1., 2., 3.], [4., 5., 6.]])
+    print *, shape([[1., 2., 3.], [4., 5.]])
+    print *, shape([[1., 2.], [4., 5., 6.]])
+    print *, shape(42)
+end program
```
Comment (HarshitaKalani) [inline integration_tests/intrinsics_172.f90:6]: Can you add checks using `if` and `error stop` to ensure it is computing correctly?

**Review by certik — APPROVED** [review]

**Comment by certik** [comment]
Conflicts must be resolved.

**Comment by certik** [comment]
@parth121101 can you please resolve the conflicts one more time.

**Comment by certik** [comment]
Please remove the binary `src/bin/main_lfortran` (!) and squash this into just one commit.

**Review by certik — APPROVED** [review]

**Pull request merged**